src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -579,15 +586,61 @@ def get_workflow_run_logs(self, wf_run_id: _models.WorkflowRunID) -> bytes:
             ).dict(),
         )
 
+        # Handle errors
         if resp.status_code == codes.NOT_FOUND:
             raise _exceptions.WorkflowRunLogsNotFound(wf_run_id)
         elif resp.status_code == codes.BAD_REQUEST:
             raise _exceptions.InvalidWorkflowRunID(wf_run_id)
 
         _handle_common_errors(resp)
 
-        # TODO: unzip, get logs (ORQSDK-652)
-        return resp.content
+        # Decompress data
+        try:
+            unzipped: bytes = zlib.decompress(resp.content, 16 + zlib.MAX_WBITS)
+        except zlib.error as e:
+            raise _exceptions.WorkflowRunLogsNotReadable(wf_run_id) from e
+
+        untarred = TarFile(fileobj=io.BytesIO(unzipped)).extractfile("step-logs")
+        assert untarred is not None
+        decoded = r"""{}""".format(untarred.read().decode("utf-8"))
+
+        # Parse the decoded data as logs
+        # TODO: index by taskinvocationID rather than workflowrunID [ORQSDK-777]
+        logs = []
+        for section in decoded.split("\n"):
+            if len(section) < 1:
+                continue
+            for log in json.loads(section):
+                try:
+                    # Orquestra logs are jsonable - where we can we parse these and
+                    # extract the useful information
+                    parsed_log_message = json.loads(log[1]["log"])
+                    interpreted_log = WFLog(
+                        timestamp=datetime.fromtimestamp(log[0], timezone.utc),
+                        level=parsed_log_message["level"],
+                        filename=parsed_log_message["filename"],
+                        message=parsed_log_message["message"],
+                        wf_run_id=parsed_log_message["wf_run_id"],
+                        task_inv_id=None,
+                        task_run_id=parsed_log_message["task_run_id"],
+                    )
```

Comment:
Let's use one of your examples.
```json
[
  1678804402.015901,
  {
    "tag":"workflow.logs.ray.wf_def-FfnTz-r000",
    "ray_filename":"/tmp/ray/session_latest/logs/worker-a601931abcd04110b31760f245e4c61d1cbd00f5943fabd5eeca1d9e-01000000-235.err",
    "log":"{\"timestamp\": \"2023-03-14T14:32:57.935535+00:00\", \"level\": \"ERROR\", \"filename\": \"_dag.py:195\", \"message\": \"Traceback (most recent call last):\\n  File \\\"/home/orquestra/venv/lib/python3.9/site-packages/orquestra/sdk/_ray/_dag.py\\\", line 192, in _ray_remote\\n    return wrapped(*inner_args, **inner_kwargs)\\n  File \\\"/home/orquestra/venv/lib/python3.9/site-packages/orquestra/sdk/_ray/_dag.py\\\", line 147, in __call__\\n    return self._fn(*unpacked_args, **unpacked_kwargs)\\n  File \\\"/Users/benjaminmummery/Documents/Projects/orquestra-workflow-sdk/../scratch/scratch.py\\\", line 12, in do_thing\\nException: Blah\\n\", \"wf_run_id\": \"wf.wf_def.3c5f938\", \"task_run_id\": \"invocation-0-task-do-thing\"}"
  }
]
```

The `"log"` above should 1:1 correspond to `WFLog`* (that's the purpose of `WFLog` class), so we can replace all the JSON accesses with just:
```suggestion
                    wf_log_json = log[1]["log"]
                    interpreted_log = WFLog.parse_json(wf_log_json)
```
Pydantic models excel at parsing JSON strings! Bonus point: this fixes the bug with `task_inv_id=None,` (most log lines should have it + we're using this field for filtering later on)

(*) disclaimer: the above is true for log lines produced by [our structured logger](https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/f5defc69058644d2044a0de04f23447b3e7cda39/src/orquestra/sdk/_base/_log_adapter.py#L20)

Reply:
Ah, that's much nicer, thanks!

Reply:
[3ad8f0e](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/79/commits/3ad8f0e1781a4459dc2182c800ca3095b9d128ad)